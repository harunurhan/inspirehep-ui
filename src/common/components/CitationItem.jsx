import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { Map } from 'immutable';
import { List } from 'antd';
import { Link } from 'react-router-dom';

import AuthorList from './AuthorList';
import Latex from '../../common/components/Latex';
import PublicationInfoList from './PublicationInfoList';

class CitationItem extends Component {
  static renderTitle(citation) {
    const recordId = citation.get('control_number');
    const title = citation.getIn(['titles', 0, 'title'], '');
    return (
      <Link to={`/literature/${recordId}`}>
        <Latex>{title}</Latex>
      </Link>
    );
  }

  render() {
    const { citation } = this.props;
    const publicationInfo = citation.get('publication_info');
    return (
      <List.Item>
        <List.Item.Meta
          title={CitationItem.renderTitle(citation)}
          description={
            <Fragment>
              <AuthorList
                recordId={citation.get('control_number')}
                authors={citation.get('authors')}
              />
              <PublicationInfoList
                publicationInfo={publicationInfo}
                labeled={false}
              />
            </Fragment>
          }
        />
      </List.Item>
    );
  }
}

CitationItem.propTypes = {
  citation: PropTypes.instanceOf(Map).isRequired,
};

export default CitationItem;
