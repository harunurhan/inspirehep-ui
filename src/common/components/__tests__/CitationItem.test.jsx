import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';

import CitationItem from '../CitationItem';

describe('CitationItem', () => {
  it('renders with full citation', () => {
    const citation = fromJS({
      titles: [{ title: 'Title' }],
      control_number: 12345,
      authors: [{ full_name: 'Author' }],
      publication_info: [
        {
          journal_title: 'Journal',
        },
      ],
    });
    const wrapper = shallow(<CitationItem citation={citation} />);
    expect(wrapper).toMatchSnapshot();
  });
});
