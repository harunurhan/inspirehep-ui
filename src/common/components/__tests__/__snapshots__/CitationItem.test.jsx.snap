// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CitationItem renders with full citation 1`] = `
<Item>
  <Meta
    description={
      <Unknown>
        <AuthorList
          authors={
            Immutable.List [
              Immutable.Map {
                full_name: "Author",
              },
            ]
          }
          enableShowAll={false}
          forSupervisors={false}
          limit={5}
          recordId={12345}
          total={-1}
          wrapperClassName={null}
        />
        <PublicationInfoList
          labeled={false}
          publicationInfo={
            Immutable.List [
              Immutable.Map {
                journal_title: "Journal",
              },
            ]
          }
        />
      </Unknown>
    }
    title={
      <Link
        replace={false}
        to="/literature/12345"
      >
        <Latex>
          Title
        </Latex>
      </Link>
    }
  />
</Item>
`;
